src/Symfony/Contracts/HttpClient/HttpClientInterface.php

```diff
@@ -0,0 +1,82 @@
+                                //   an array is passed, it is meant as a form payload of field names and values
+        'json' => null,         // array|\JsonSerializable - when set, implementations MUST set the "body" option to
+                                //   the JSON-encoded value and set the "content-type" headers to a JSON-compatible
+                                //   value it is they are not defined - typically "application/json"
+        'max_redirects' => 20,  // int - the maximum number of redirects to follow - any "Authorization" headers MUST
+                                //       NOT follow except for the initial host name as given in the original URL
+        'http_version' => null, // string - defaults to the best supported version, typically 1.1 or 2.0
+        'base_uri' => null,     // string - the URI to resolve relative URLs, following rules in RFC 3986, section 2
+        'buffer' => true,       // bool - whether the content of the response should be buffered or not
+        'on_progress' => null,  // callable - called on download/upload at least once per second with arguments
+                                //            int $dlNow, int $dlSize and array $info; any exceptions thrown from the
+                                //            callback MUST abort the request
+        'resolve' => [],        // string[] - a map of host to IP address that SHOULD replace DNS resolution
+        'proxy' => null,        // string - by default, the proxy-related env vars handled by curl SHOULD be honored
+        'timeout' => null,      // float - the inactivity timeout - defaults to ini_get('default_socket_timeout')
+        'bindto' => '0',        // string - the interface or the local socket to bind to
+        'verify_peer' => true,  // see https://php.net/context.ssl for the following options
+        'verify_host' => true,
+        'cafile' => null,
+        'capath' => null,
+        'local_cert' => null,
+        'local_pk' => null,
+        'passphrase' => null,
+        'ciphers' => null,
+        'peer_fingerprint' => null,
+        'capture_peer_cert_chain' => false,
+    ];
+
+    /**
+     * Requests an HTTP resource.
+     *
+     * Responses SHOULD be lazy, but their status code MUST be
+     * checked even if none of their public methods are called.
+     *
+     * @param string   $url     The URL of the resource
+     * @param iterable $options The options of the request
+     *
+     * @throws TransportExceptionInterface When an unsupported option is passed or
+     *                                     if any error happens at the transport layer
+     */
+    public function request(string $method, string $url, iterable $options = []): ResponseInterface;
+
+    /**
+     * Yields responses chunk by chunk as they complete.
+     *
+     * @param ResponseInterface[]|ResponseInterface $responses One or more responses created by the current HTTP client
```

Comment:
Shouldn't we specify here that an iterable of `ResponseInterface` is OK ?

Reply:
Isn't what this already tells?

Reply:
For me it's not clear, `ResponseInterface[]` = an array of `ResponseInterface`.

In other components, we added the `iterable` in the doc type for similar cases. Since it's the contract part, I think it's really important to be clear that the implementation should handle an `iterable` of `ResponseInterface`.

Reply:
done